tests/test_codebase/test_mmcls/test_classification.py

```diff
@@ -73,58 +81,57 @@ def test_init_pytorch_model(from_mmrazor: Any):
     assert isinstance(model, BaseClassifier)
 
 
-@pytest.fixture
-def backend_model():
+@pytest.fixture(scope='module')
+def backend_model(task_processor):
     from mmdeploy.backend.onnxruntime import ORTWrapper
-    ort_apis.__dict__.update({'ORTWrapper': ORTWrapper})
-    wrapper = SwitchBackendWrapper(ORTWrapper)
-    wrapper.set(outputs={
-        'output': torch.rand(1, num_classes),
-    })
+    with SwitchBackendWrapper(ORTWrapper) as wrapper:
+        wrapper.set(outputs={
+            'output': torch.rand(1, num_classes),
+        })
 
-    yield task_processor.init_backend_model([''])
-
-    wrapper.recover()
+        yield task_processor.init_backend_model([''])
 
 
 def test_init_backend_model(backend_model):
     assert isinstance(backend_model, torch.nn.Module)
 
 
-def test_create_input():
-    inputs = task_processor.create_input(img, input_shape=img_shape)
+@pytest.fixture(scope='module')
+def model_inputs(task_processor, img):
+    return task_processor.create_input(img, input_shape=img_shape)
+
+
+def test_create_input(model_inputs):
+    inputs = model_inputs
     assert isinstance(inputs, tuple) and len(inputs) == 2
 
 
-def test_run_inference(backend_model):
-    input_dict, _ = task_processor.create_input(img, input_shape=img_shape)
+def test_run_inference(task_processor, backend_model, model_inputs):
+    input_dict, _ = model_inputs
     results = task_processor.run_inference(backend_model, input_dict)
     assert results is not None
 
 
-def test_visualize(backend_model):
-    input_dict, _ = task_processor.create_input(img, input_shape=img_shape)
+def test_visualize(task_processor, backend_model, tmp_path, img, model_inputs):
+    input_dict, _ = model_inputs
     results = task_processor.run_inference(backend_model, input_dict)
-    with TemporaryDirectory() as dir:
-        filename = dir + 'tmp.jpg'
-        task_processor.visualize(backend_model, img, results[0], filename, '')
-        assert os.path.exists(filename)
+    filename = str(tmp_path / 'tmp.jpg')
```

Comment:
is it OK to concat string in this way? Better use `os.path` to join directories.

Reply:
[tmp_path](https://docs.pytest.org/en/7.2.x/reference/reference.html#tmp-path) is a builtin fixture in pytest which return a [pathlib.Path](https://docs.python.org/3/library/pathlib.html#pathlib.Path) object.